cpp/cmake/apache-arrow.cmake

```diff
@@ -107,11 +108,11 @@ function(build_arrow)
 
     find_package(Threads)
     find_package(Arrow QUIET)
-    set(ARROW_VERSION_TO_BUILD "12.0.0" CACHE INTERNAL "arrow version")
-    if (Arrow_FOUND) # arrow is installed, build the same version as the installed one
-        message(STATUS "Found Arrow installed, align to version: ${Arrow_VERSION}")
-        set(ARROW_VERSION_TO_BUILD "${Arrow_VERSION}" CACHE INTERNAL "arrow version")
-    endif ()
+    set(ARROW_VERSION_TO_BUILD "15.0.0" CACHE INTERNAL "arrow version")
+    # if (Arrow_FOUND) # arrow is installed, build the same version as the installed one
+        # message(STATUS "Found Arrow installed, align to version: ${Arrow_VERSION}")
+        # set(ARROW_VERSION_TO_BUILD "${Arrow_VERSION}" CACHE INTERNAL "arrow version")
+    # endif ()
     set(GAR_ARROW_SOURCE_FILE "https://www.apache.org/dyn/closer.lua?action=download&filename=arrow/arrow-${ARROW_VERSION_TO_BUILD}/apache-arrow-${ARROW_VERSION_TO_BUILD}.tar.gz")
```

Comment:
Can environment variables or parameters be passed here?

Reply:
If the user has their own download link, they can identify and determine the version after installation.

> Can environment variables or parameters be passed here?

Reply:
hi, @jasinliu, thanks for the advice. Maybe we can not expose the variable to user, and if there is arrow installed in the system, we can align to the version

Reply:
> If the user has their own download link, they can identify and determine the version after installation.
> 
> > Can environment variables or parameters be passed here?

and I not sure what you mean own download link? you mean there maybe arrow that modified by user?

Reply:
> > If the user has their own download link, they can identify and determine the version after installation.
> > > Can environment variables or parameters be passed here?
> 
> and I not sure what you mean own download link? you mean there maybe arrow that modified by user?

For example, the specified mirror link， the modification of the arrow，or some test version of the arrow. Custom links can provide some flexibility。

Reply:
> > > If the user has their own download link, they can identify and determine the version after installation.
> > > > Can environment variables or parameters be passed here?
> > 
> > 
> > and I not sure what you mean own download link? you mean there maybe arrow that modified by user?
> 
> For example, the specified mirror link， the modification of the arrow，or some test version of the arrow. Custom links can provide some flexibility。

But we are supposed to only support the original version of arrow, provide a link for that is a little weird. I suggest we still use the link from apache, and if user want to use other link, he could modify the cmake file directly.